### PR TITLE
fix: eslint errors

### LINKED ISSUE #242: fix: eslint errors
**Node.js version**: v12.16.1
 **itoolkit version**: 1.0.0-alpha.1 (latest master)
 **IBM i version**: 7.2
 **XMLSERVICE version**: XML Toolkit 2.0.2-dev

**Describe the bug**
We have some eslint erros that we need to cleanup.

**To Reproduce**
1. clone the repository
2. Run `$ npm install`
3. Run `$ ./node_modules/eslint/bin/eslint.js lib/ test/`

image

**Expected behavior**
We should have 0 eslint errors

### PR DESCRIPTION
**Pull request by abmusse**
Fixes #242

### CODE REVIEW

File: test/functional/checkObjectExists.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 const lib = 'NODETKTEST';
 const createLib = `CRTLIB LIB(${lib}) TYPE(*TEST) TEXT('Used to test Node.js toolkit')`;
 const findLib = `SELECT SCHEMA_NAME FROM qsys2.sysschemas WHERE SCHEMA_NAME = '${lib}'`;
-const dataQueue = 'TESTQ';
```
Comment (abmusse) [inline test/functional/checkObjectExists.js:4]: ../nodejs-itoolkit/test/functional/checkObjectExists.js
    4:7   error  'dataQueue' is assigned a value but never used                                   no-unused-vars

File: test/functional/deprecated/iProdFunctional.js [hunk 2]
```diff
@@ -132,7 +132,6 @@ describe('iProd Functional Tests', () => {
   // REST transport currently failing with 414 URI Too Long response code
   // The requested URL's length exceeds the capacity limit for this server
   describe('getInstalledProducts', () => {
-    // eslint-disable-next-line func-names
```
Comment (abmusse) [inline test/functional/deprecated/iProdFunctional.js:135]: No need for this.

File: test/functional/iDataQueueFunctional.js [hunk 3]
```diff
@@ -34,8 +34,8 @@ describe('DataQueue Functional Tests', () => {
     checkObjectExists(config, dqName, '*DTAQ', (error) => {
       if (error) { throw error; }
 
-      checkObjectExists(config, dqName2, '*DTAQ', (error) => {
-        if (error) { throw error; }
```
Comment (abmusse) [inline test/functional/iDataQueueFunctional.js:38]: Fixes

```
../nodejs-itoolkit/test/functional/iDataQueueFunctional.js
  37:52  error  'error' is already declared in the upper scope  no-shadow
```

File: lib/transports/idbTransport.js [hunk 4]
```diff
@@ -101,7 +102,6 @@ const idbCall = (config, xmlInput, cb) => {
             xmlOutput += chunk.OUT151;
           });
           done(null, xmlOutput);
-          return;
```
Comment (abmusse) [inline lib/transports/idbTransport.js:104]: Fixes

```
../nodejs-itoolkit/lib/transports/idbTransport.js
  104:11  error  Unnecessary return statement             no-useless-return
```

File: .eslintrc.js [hunk 5]
```diff
@@ -1 +1,13 @@
...
+module.exports =  { 
+    "extends": "airbnb-base",
+    env: {
+        node: true
+    },
+    rules: {
+        // Tweak rules set by airbnb config
+        'no-console': 'off', // allow console
+        'import/no-extraneous-dependencies': ["error", {"optionalDependencies": true}], // Allow optional dep import
+        // Warn unreoslved imports ie: idb-connector import will be unresolved when on non IBM i system.
```
Comment (kadler) [inline .eslintrc.js:10]:
```suggestion
        // Warn unresolved imports ie: idb-connector import will be unresolved when on non IBM i system.
```

**Pull request merged**